File: cybersphere/CyberSphere/node_app/src/auth/sessionAuth.js
function getUserId(req) {
  if (!req.session) return null;
  const v = req.session.userId;
  if (v === undefined || v === null) return null;
  if (typeof v === 'number') return v;
  const n = Number(v);
  return Number.isFinite(n) ? n : null;
}

function isAdmin(req) {
  if (!req.session) return false;
  return String(req.session.isAdmin || '') === 'true' || req.session.isAdmin === true;
}

module.exports = { getUserId, isAdmin };

File: cybersphere/CyberSphere/src/main/java/com/cybersphere/cybersphere/security/SessionAuth.java
package com.cybersphere.cybersphere.security;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpSession;

public class SessionAuth {
  public static Long getUserId(HttpServletRequest req) {
    HttpSession s = req.getSession(false);
    if (s == null) return null;
    Object v = s.getAttribute("userId");
    if (v == null) return null;
    if (v instanceof Long) return (Long) v;
    if (v instanceof Integer) return ((Integer) v).longValue();
    try {
      return Long.parseLong(String.valueOf(v));
    } catch (Exception e) {
      return null;
    }
  }

  public static boolean isAdmin(HttpServletRequest req) {
    HttpSession s = req.getSession(false);
    if (s == null) return false;
    Object v = s.getAttribute("isAdmin");
    return v != null && "true".equalsIgnoreCase(String.valueOf(v));
  }
}

